src/coreclr/jit/morph.cpp

```diff
@@ -9042,6 +9042,11 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
         return tree;
     }
 
+    if (tree->IsNothingNode())
```

Comment:
Is this actually reachable? Presumably, `NOP`s should go through `fgMorphLeaf` now.

Reply:
Yes, `gtFoldExpr` called above may return a NOP (to be precise, `gtFoldExprConst` does that for:
```
*  BOUNDS_CHECK_Rng void   <l:$285, c:$284>
+--*  CNS_INT   int    0 $40
\--*  CNS_INT   int    1 $41
```